cmd/notation/verify.go

```diff
@@ -103,26 +117,19 @@ func resolveReference(command *cobra.Command, opts *verifyOpts) (registry.Refere
 		return registry.Reference{}, err
 	}
 
-	if isDigestReference(opts.reference) {
+	// reference is a digest reference
+	if ref.ValidateReferenceAsDigest() == nil {
 		return ref, nil
 	}
 
-	// Resolve tag reference to digest reference.
-	manifestDesc, err := getManifestDescriptorFromReference(command.Context(), &opts.SecureFlagOpts, opts.reference)
+	// resolve tag to digest reference
+	manifestDesc, ref, err := getManifestDescriptorFromReference(command.Context(), &opts.SecureFlagOpts, opts.reference)
```

Comment:
empty reference (i.e. no tag and not digest) is not checked.

Reply:
Refactored verify.go.